baseapp-chats/baseapp_chats/graphql/subscriptions.py

```diff
@@ -96,25 +96,32 @@ class ChatRoomOnMessage(channels_graphql_ws.Subscription):
 
     class Arguments:
         room_id = graphene.ID(required=True)
+        profile_id = graphene.ID(required=True)
 
     @staticmethod
-    def subscribe(root, info, room_id):
-        room = database_sync_to_async(get_obj_from_relay_id)(info, room_id)
-
+    async def subscribe(root, info, room_id, profile_id):
+        room = await database_sync_to_async(get_obj_from_relay_id)(info, room_id)
         user = info.context.channels_scope["user"]
+        profile = await database_sync_to_async(get_obj_from_relay_id)(info, profile_id)
+
         if not user.is_authenticated or not database_sync_to_async(user.has_perm)(
-            "baseapp_chats.view_chatroom", room
+            "baseapp_profiles.use_profile", profile
         ):
             return []
+
+        if not database_sync_to_async(room.participants.filter(profile=profile).exists)():
```

Comment:
please lets change this to be like before:

```suggestion
        if not database_sync_to_async(user.has_perm)(
            "baseapp_chats.view_chatroom", room
        )
```

it not only checks if the person is a member but also if he is blocked or not

Reply:
@nossila: I am not sure whether we necessarily need to check this again since it is only for real-time updates about sent messages. When a message is sent, we check for permission (including blocks), so I think it'd be reasonable to assume that we can always synchronize messages sent on the server with the client without checking again. Also `user.has_perm` only checks whether some profile of accessible to `user` is in the chatroom, but not the given `profile`. I'll switch to 
```
if not database_sync_to_async(room.participants.filter(profile=profile).exists)() 
    or not database_sync_to_async(user.has_perm)("baseapp_chats.view_chatroom", room):
...
```

Reply:
I think it would be good to rethink some of these permissions. E.g. all chat features are related to profiles, not users. So permissions like `view_chatroom` should use the `profile`, not just `user`. Also, I am currently not allowed to view a chat if there is any user blocked by me on a chat. For large chat groups, this might not make much sense; I might even lose access to all previous messages from a chat if such a user joins the group.